src/Giraffe/Routing.fs

```diff
@@ -10,37 +10,97 @@ open FSharp.Control.Tasks.V2.ContextInsensitive
 open Giraffe.FormatExpressions
 
 // ---------------------------
-// Private sub route helper functions
+// Sub Routing Feature
 // ---------------------------
 
-[<RequireQualifiedAccess>]
-module SubRouting =
+[<Literal>]
+let private RouteKey = "giraffe_route"
+
+type ISubRoutingFeature =
+    abstract member CompatibilityMode : GiraffeCompatibilityMode with get
+
+    abstract member GetResolvedPath   : unit   -> string option
+    abstract member SetResolvedPath   : string -> unit
+    abstract member ClearResolvedPath : unit   -> unit
+
+type SubRoutingFeature (compatibilityMode : GiraffeCompatibilityMode) =
```

Comment:
If this implementation has the compatibility mode, why not have the middleware that creates this pass in HttpContext too? Then you can have these implementations handle setting the old Items based key.